scripts/test.js

```diff
@@ -4,90 +4,104 @@ require("@app/config/env");
 const { execSync, spawnSync: rawSpawnSync } = require("child_process");
 const concurrently = require("concurrently");
 
-const spawnSync = (cmd, args, options) => {
-  const result = rawSpawnSync(cmd, args, {
-    stdio: ["pipe", "inherit", "inherit"],
-    env: {
-      ...process.env,
-      YARN_SILENT: "1",
-      npm_config_loglevel: "silent",
-    },
-    ...options,
-  });
+// Dear graphile-migrate, please treat the test DB as if it were the shadow DB
+process.env.SHADOW_DATABASE_URL = process.env.TEST_DATABASE_URL;
 
-  const { error, status, signal, stderr, stdout } = result;
+// Signal to graphile-migrate scripts that we're in the tests
+process.env.IN_TESTS = "1";
+process.env.NODE_ENV = "test";
 
-  if (error) {
-    throw error;
+const cmdArgs = process.argv.slice(2);
+const watchMode = cmdArgs.find(
+  (arg) => arg === "--watch" || arg === "--watchAll"
+);
+const delayArg = cmdArgs.indexOf("--delay");
+let delaySeconds = null;
+if (delayArg > -1) {
+  delaySeconds = parseFloat(cmdArgs[delayArg + 1]);
+  if (isNaN(delaySeconds)) {
+    throw new Error("Did not get a valid delay argument in seconds.");
   }
+  setTimeout(main, delaySeconds * 1000);
+} else {
+  main();
+}
 
-  if (status || signal) {
-    if (stdout) {
-      console.log(stdout.toString("utf8"));
-    }
-    if (stderr) {
-      console.error(stderr.toString("utf8"));
-    }
-    if (status) {
-      throw new Error(
-        `Process exited with status '${status}' (running '${cmd} ${
-          args ? args.join(" ") : ""
-        }')`
-      );
-    } else {
-      throw new Error(
-        `Process exited due to signal '${signal}' (running '${cmd} ${
-          args ? args.join(" ") : null
-        }')`
-      );
-    }
-  }
+function main() {
+  const spawnSync = (cmd, args, options) => {
+    const result = rawSpawnSync(cmd, args, {
+      stdio: ["pipe", "inherit", "inherit"],
+      env: {
+        ...process.env,
+        YARN_SILENT: "1",
+        npm_config_loglevel: "silent",
+      },
+      shell: true,
+      ...options,
+    });
```

Comment:
There's no need for this function to be defined in this scope (I think?), please move it back to the root scope.

Reply:
It was necessary as it was defined as it could not find the anonymous function in the main scope. But it also works if we change to a normal function declaration.